crates/bevy_math/src/curves/fixed.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    curves::{Curve, CurveCursor},
+    interpolation::Lerp,
+};
+
+// TODO: impl Serialize, Deserialize
+/// Curve with evenly spaced keyframes, in another words a curve with a fixed frame rate.
+///
+/// This curve maintains the faster sampling rate over a wide range of frame rates, because
+/// it doesn't rely on keyframe cursor. As a downside, it will have a bigger memory foot print.
+#[derive(Default, Debug)]
+pub struct CurveFixed<T> {
+    frame_rate: f32,
+    /// Negative number of frames before the curve starts
+    negative_offset: f32,
+    pub keyframes: Vec<T>,
+}
+
+impl<T: Clone> Clone for CurveFixed<T> {
```

Comment:
Is it not possible to just `#[derive(Clone)]` on the original struct?

Reply:
Apparently yes!